base/point.h

```diff
@@ -0,0 +1,51 @@
+#ifndef _BASE_POINT_H_
+#define _BASE_POINT_H_
+#include <string>
+
+/**
+ * \file point.h
+ * \brief Header defining the multi-dimensional point interface.
+ */
+
+/**
+ * Basic objects to be used by all machine learning algorithms.
+ */
+namespace base {
+
+/**
+ * \brief Multi-dimensional object interface.
+ *
+ * This interface is the primitive input data type for the learning algorithms.
+ * For every new type of object we want to learn from, we must implement a
+ * suitable concrete class to handle it.
+ */
+class Point {
+ public:
+  /**
+   * Default constructor.
+   */
+  virtual Point() = 0;
+  /** String parsing constructor.
+   *
+   * \param point_string should be equal to the return value of ToString().
+   */
+  virtual Point(const std::string& point_string) = 0;
```

Comment:
Done.
For the question: we will pretty much always be reading entire lines of the input files, so it would make sense to read entire chunks at a time and pass them directly to a constructor. What do you think?